pgmpy/utils/state_name.py

```diff
@@ -0,0 +1,201 @@
+"""
+The file contains decorators to manage the user defined
+variable state names. It maps the internal representaion
+of the varibale states to the user defined state names and
+vice versa.
+"""
+
+
+class StateNameInit():
+    """
+    The class behaves as a decorator for __init__ methods.
+    It adds a dictionary as an attribute to the various classes
+    where mapping is required for the variable state names.
+    The dictionary has the following format -
+    {'x1': ['on', 'off']}
+    where, 'x1' is a variable and 'on', 'off' are its correspinding
+    states.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> from pgmpy.factors import Factor
+    >>> sn = {'speed': ['low', 'medium', 'high'],
+             'switch': ['on', 'off'],
```

Comment:
@yashu-seth Fix this please. You need to start a continuation line with triple dots in doctests. Check other files for example.